front/lib/resources/agent_message_feedback_resource.ts

```diff
@@ -84,24 +75,63 @@ export class AgentMessageFeedbackResource extends BaseResource<AgentMessageFeedb
     );
   }
 
-  static async fetchByUserAndAgentMessages(
-    user: UserType,
-    agentMessages: AgentMessage[]
-  ): Promise<AgentMessageFeedbackResource[]> {
+  static async fetchByAgentConfigurationId({
+    auth,
+    agentConfigurationId,
+    pagination,
+  }: {
+    auth: Authenticator;
+    agentConfigurationId: string;
+    pagination: {
+      limit: number;
+      olderThan?: Date;
+    };
+  }): Promise<AgentMessageFeedback[]> {
     const agentMessageFeedback = await AgentMessageFeedback.findAll({
       where: {
-        userId: user.id,
-        agentMessageId: {
-          [Op.in]: agentMessages.map((m) => m.id),
-        },
+        agentConfigurationId,
+        // Necessary for global models who share ids across workspaces
+        workspaceId: auth.getNonNullableWorkspace().id,
+        ...(pagination.olderThan && {
+          createdAt: {
+            [Op.lt]: pagination.olderThan,
+          },
+        }),
       },
+
+      include: [
+        {
+          model: AgentMessageModel,
+          attributes: ["id"],
+          as: "agentMessage",
+          include: [
+            {
+              model: Message,
+              as: "message",
+              attributes: ["id", "sId"],
+              include: [
+                {
+                  model: Conversation,
```

Comment:
woow is this properly indexed and fast? Do we really need all that?

Reply:
A triple JOIN can't be good news can it?

Reply:
Hmm I didn't see any more direct way In practice.
How is doing 3 db round trips more optimized than joining ? I checked and all these fields are indexed

Reply:
I agree in principle. The problem is practice 👍